package/MDAnalysis/lib/_cutil.pyx

```diff
@@ -51,7 +53,7 @@ def unique_int_1d(np.ndarray[np.int64_t, ndim=1] values):
     cdef int i = 0
     cdef int j = 0
     cdef int n_values = values.shape[0]
-    cdef np.ndarray[np.int64_t, ndim=1] result = np.empty(n_values, dtype=np.int64)
+    cdef long[:] result = np.empty(n_values, dtype=np.int64)
```

Comment:
you should continue to use the `np.int64_t` type to ensure the size of the type independent of the used platform. What a long  can vary in size.